src/Sylius/Bundle/TaxonomyBundle/Form/Type/TaxonChoiceType.php

```diff
@@ -107,4 +100,25 @@ public function getBlockPrefix()
     {
         return 'sylius_taxon_choice';
     }
+
+    /**
+     * @param null|string   $root
+     * @param null|callable $filter
+     *
+     * @return array|TaxonInterface[]
+     */
+    private function getTaxons($root = null, $filter = null)
+    {
+        if (null !== $root) {
+            $taxons = $this->taxonRepository->findChildren($root, true);
+        } else {
```

Comment:
What about adding a nullable argument `$rootCode` to method `findNodesTreeSorted` ? It means putting some logic in the repository but we only need an `if` without `else`.